modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/appcatalog/ApplicationDeploymentRepository.java

```diff
@@ -0,0 +1,159 @@
+        }
+        if (applicationDeploymentEntity.getPreJobCommands() != null) {
+            applicationDeploymentEntity.getPreJobCommands().forEach(prejobCommandEntity -> prejobCommandEntity.setAppdeploymentId(applicationDeploymentId));
+        }
+        if (applicationDeploymentEntity.getPostJobCommands() != null) {
+            applicationDeploymentEntity.getPostJobCommands().forEach(postjobCommandEntity -> postjobCommandEntity.setAppdeploymentId(applicationDeploymentId));
+        }
+        if (applicationDeploymentEntity.getLibPrependPaths() != null) {
+            applicationDeploymentEntity.getLibPrependPaths().forEach(libraryPrependPathEntity -> libraryPrependPathEntity.setDeploymentId(applicationDeploymentId));
+        }
+        if (applicationDeploymentEntity.getLibAppendPaths() != null) {
+            applicationDeploymentEntity.getLibAppendPaths().forEach(libraryApendPathEntity -> libraryApendPathEntity.setDeploymentId(applicationDeploymentId));
+        }
+        if (applicationDeploymentEntity.getSetEnvironment() != null) {
+            applicationDeploymentEntity.getSetEnvironment().forEach(appEnvironmentEntity -> appEnvironmentEntity.setDeploymentId(applicationDeploymentId));
+        }
+        if (!isAppDeploymentExists(applicationDeploymentId))
+            applicationDeploymentEntity.setCreationTime(new Timestamp(System.currentTimeMillis()));
+        applicationDeploymentEntity.setUpdateTime(new Timestamp(System.currentTimeMillis()));
+        return execute(entityManager -> entityManager.merge(applicationDeploymentEntity));
+    }
+
+    @Override
+    public String addApplicationDeployment(ApplicationDeploymentDescription applicationDeploymentDescription, String gatewayId) throws AppCatalogException {
+        return saveApplicationDeploymentDescriptorData(applicationDeploymentDescription, gatewayId);
+    }
+
+    @Override
+    public void updateApplicationDeployment(String deploymentId, ApplicationDeploymentDescription updatedApplicationDeploymentDescription) throws AppCatalogException {
+        saveApplicationDeploymentDescriptorData(updatedApplicationDeploymentDescription, null);
+    }
+
+    @Override
+    public ApplicationDeploymentDescription getApplicationDeployement(String deploymentId) throws AppCatalogException {
+        Map<String, Object> queryParameters = new HashMap<>();
+        queryParameters.put(DBConstants.ApplicationDeployment.APPLICATION_DEPLOYMENT_ID, deploymentId);
+        ApplicationDeploymentDescription applicationDeploymentDescription = select(QueryConstants.FIND_APPLICATION_DEPLOYMENT, -1, 0, queryParameters).get(0);
+        return applicationDeploymentDescription;
+    }
+
+    @Override
+    public List<ApplicationDeploymentDescription> getApplicationDeployements(Map<String, String> filters) throws AppCatalogException {
+        if(filters.containsKey(DBConstants.ApplicationDeployment.APPLICATION_MODULE_ID)) {
```

Comment:
Add debug logs in each block and add a newline at the end of each block to improve the readability